GVRf/Framework/src/org/gearvrf/scene_objects/GVRVideoSceneObject.java

```diff
@@ -167,8 +189,16 @@ public MediaPlayer getMediaPlayer() {
      * @param mediaPlayer
      *            An Android {@link MediaPlayer}
      */
-    public void setMediaPlayer(MediaPlayer mediaPlayer) {
-        mVideo.setMediaPlayer(mediaPlayer);
+    public void setMediaPlayer(final MediaPlayer mediaPlayer) {
+        if (mVideo == null) {
```

Comment:
Because GVRVideoSceneObject requires a MediaPlayer in its constructor, this would introduce two ways of doing the same thing. That's something not optimal from API point of view.